gcloud-java-core/src/main/java/com/google/gcloud/PageImpl.java

```diff
@@ -38,6 +40,37 @@
     Page<T> nextPage();
   }
 
+  /**
+   * Base class for page fetchers.
+   *
+   * @param <T> the value type that the page holds. {@code T} should implement
+   *     {@code java.io.Serializable}
+   * @param <O> the service options type
+   */
+  @SuppressWarnings("rawtypes")
+  public abstract static class BasePageFetcher<T, O extends ServiceOptions>
+      implements NextPageFetcher<T> {
+
+    private static final long serialVersionUID = -3333056232729661990L;
+    protected final Map<?, ?> requestOptions;
+    protected final O serviceOptions;
+
+    protected BasePageFetcher(O serviceOptions, Enum<?> pageTokenOption,
```

Comment:
Fine as is, though to make it more generic we could make `pageTokenOption` an `Object` and compare bellow using `Objects.equals`